ui/src/components/features/build-panel/components/select-files.tsx

```diff
@@ -1,74 +1,73 @@
 import { Badge } from "@/components/ui/badge";
 import { FormControl, FormField, FormItem } from "@/components/ui/form";
 import Spinner from "@/components/ui/spinner";
-import { useFiles } from "@/data-provider/query-service";
+import { useToast } from "@/components/ui/use-toast";
+import { useAssistantFiles, useDeleteAssistantFile, useDeleteFile, useFiles } from "@/data-provider/query-service";
+import { useSlugRoutes } from "@/hooks/useSlugParams";
 import { CircleX } from "lucide-react";
 import { useEffect, useState } from "react";
 import { UseFormReturn, useFieldArray } from "react-hook-form";
 
-type TSelectFilesProps = {
-  form: UseFormReturn<any>;
-};
-
 type TBadgeValue = {
   label: string;
   value: string;
 };
 
-export default function SelectFiles({ form }: TSelectFilesProps) {
-  const [badgeValues, setBadgeValues] = useState([]);
+export default function SelectFiles() {
+  const [badgeValues, setBadgeValues] = useState<TBadgeValue[]>([]);
```

Comment:
The `badgeValues` can also be derived state. No need to keep react state and useeffect for it.

Reply:
Yes! Totally - same reasoning as above. I know this, not sure why I decided to use state here.